### ISSUE TITLE
SMART Implementation

### ISSUE THREAD

**Issue by Plofstoffel** [smart-on-fhir/smart-dev-sandbox#6]
Hi,

I have been playing around with the Dev-Sandbox and would like to confirm the following.

- Where is the SMART implementation?
- Is the Docker Image in this sandbox implementing an Interceptor as per the Hapi server documentation?

Sorry for creating an issue about this but I cannot get any SMART requests to work via Postman.

**Comment by vlad-ignatov** [comment]
This is a package of several projects linked together. HAPI is user as FHIR server only. The SMART authentication layer is handled by what we call the sandbox. It is basically a NodeJS proxy server, intercepting requests to handle the authentication and then sending them upstream to the HAPI servers.

Perhaps if you share your Postman experiments I might be able to help more...

**Comment by alekline** [comment]
Hi,

Thank you for fixing this @vlad-ignatov

 - I found a couple of other places where the `HOST` variable must be added so this works everywhere:

**For the Patient Browser to work, these files need to be updated:**

- The `patient-browser` section of the `docker-compose.yml` file (line ~59)
- `smart-dev-sandbox/patient-browser/dstu2-local.tpl` (line 4)
- `smart-dev-sandbox/patient-browser/dstu3-local.tpl` (line 4)

**For the Fhir Viewer it is a bit tricky, I haven't been able to fix it...**
*I get this error: `Unknown URL origin. Consider adding your base URL to the known-servers.js file.` - but adding the server IP to the known-servers.js file still doesn't work.

**...but I think the offender files are:**
- The `patient-browser` section of the `docker-compose.yml` file (line ~71)
-  file `/usr/share/nginx/html/lib.js` (line ~253) inside the FHIR Viewer container. Note however, that this file is not exposed in the smart-sev-sandbox, you have to go inside the container to see it (using `sudo docker exec -it smart-dev-sandbox_fhir-viewer_1 /bin/sh`):
`if (!getBaseURL() && !params.url.match(/^https?:\/\/(localhost|127\.0\.0\.1)/)) {`

Again, I haven't been able to fix the Fhir Viewer, but I think the files above might be related to the problem.

**Comment by vlad-ignatov** [comment]
That is correct. I had do some additional changes to support the HOST variable. Please update.
